crates/evm/src/lib.rs

```diff
@@ -38,13 +38,62 @@ pub mod system_calls;
 /// test helpers for mocking executor
 pub mod test_utils;
 
+/// An abstraction over EVM.
+pub trait Evm {
```

Comment:
could we clarify here that this is intended to be implemented by types that wrap the `revm::Evm`